lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -0,0 +1,266 @@
+    get_es_url,
+    get_index_prefix,
+    gen_month_range,
+    get_user_term,
+)
+
+
+class DatasetsDetail(Resource):
+    """
+    Abstracted Pbench API to get detailed data for a particular
+    dataset by name.
+
+    This API a follow-on to datasets_list, which returns the names of datasets
+    for a given controller. The datasets_detail API returns details from
+    the dataset's run document.
+
+    TODO: This implementation closely follows the Javascript effect. The date
+    range given isn't used for the search directly, but rather defines the set
+    of Elasticsearch index names to search. This could be improved.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """
+        __init__ Initialize the resource with info each call will need.
+
+        Args:
+            :config: The Pbench server config object
+            :logger: a logger
+        """
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def post(self):
+        """
+        Get details for a specific Pbench dataset if it is owned by the
+        specified user:
+
+        {
+            "user": "username",
+            "name": "dataset-name",
+            "start": "start-time",
+            "end": "end-time"
+        }
+
+        "user" specifies the owner of the data to be searched; it need not
+        necessarily be the user represented by the authorization header,
+        assuming the session user is authorized to view "user"s data. This
+        will be used to constrain the Elasticsearch query.
+
+        TODO: When we have authorization infrastructure, we'll need to
+        check that "session user" has rights to view "user" data. We might
+        also default a missing "user" JSON field with the authorization
+        token's user. (Or with the default user "public" if there's no
+        token, indicating there's no logged-in user.)
+
+        "name" is the name of a Pbench agent dataset (tarball).
+
+        "start" and "end" are time strings representing a set of Elasticsearch
+        run document indices in which the dataset will be found.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, production, "staging",
+        and development server instances might all share one Elasticsearch
+        cluster by choosing unique prefixes to generate unique index
+        names.
+
+        Required headers include
+
+        Authorization:  Pbench authorization token authorized to access
```

Comment:
Perhaps word this as: `Pbench authorization token granting the caller access to the "user"'s data.`